src/stream-tcp-reassemble.c

```diff
@@ -1172,6 +1172,9 @@ static bool GetAppBuffer(const TcpStream *stream, const uint8_t **data, uint32_t
             *data_len = 0;
         }
     }
+    if (*data_len > UINT16_MAX) {
```

Comment:
This is the core of the solution !

Reply:
It's not there in the PR now. Is that on purpose?